lib/spack/spack/package.py

```diff
@@ -2162,6 +2172,27 @@ def do_uninstall(self, force=False):
         # delegate to instance-less method.
         Package.uninstall_by_spec(self.spec, force)
 
+    def do_deprecate(self, replacement, link_fn):
+        """Deprecate this package in favor of spec"""
+        # Copy metadata to deprecated directory of replacement
+        spec = self.spec
+        deprecated = spack.store.db.get_record(spec).deprecated_for
+        if deprecated:
+            if deprecated == replacement.dag_hash():
```

Comment:
It looks like nothing happens when `deprecated and deprecated != replacement.dag_hash()` - should there be a message for that?

Reply:
Yes. I had originally planned to do the future work I outlined above about re-deprecating packages, I forgot to change the error message back after attempting it.